crates/propolis-api-types/src/instance_spec/components/devices.rs

```diff
@@ -104,19 +104,44 @@ pub struct QemuPvpanic {
     // TODO(eliza): add support for the PCI PVPANIC device...
 }
 
+/// A set of settings that determines what Propolis tells guest firmware about
+/// how to boot a guest operating system.
```

Comment:
```suggestion
/// how to boot a guest.
```

minorest quibble: the settings here are processed before the guest has landed in an OS for the most part. unless there's a way we'd expect to speak to guests via UEFI (which i genuinely would believe, and would believe if there's something specific to OSes that we anticipate and i dunno about)

Reply:
I took this in a slightly different direction in 04fb943, but I think I've captured the gist of this (this is information for the firmware image to use, not--generally--the next-stage binary that it finds and transfers control to).

Reply:
oh yeah i like that a lot, thanks!